ocaml/testsuite/tests/lib-obj/uniquely_reachable_words.ml

```diff
@@ -0,0 +1,80 @@
+(* TEST
+*)
+
+let native =
+  match Sys.backend_type with
+  | Sys.Native -> true
+  | Sys.Bytecode -> false
+  | Sys.Other s -> print_endline s; assert false
+
+let sizes xs = Obj.uniquely_reachable_words (List.map Obj.repr xs)
+
+let expect_sizes xs exp_bytecode exp_native =
+  let actual = sizes xs in
+  let expected = if native then exp_native else exp_bytecode in
+  List.combine actual expected
+  |> List.iteri (fun i (a, e) ->
+    if a <> e then
+      Printf.printf "index = %i; actual = %i; expected = %i\n" i a e)
+
+type node = { id: int; used_memory: int list; mutable children: node list }
+let make id ch = { id; used_memory = List.init (Int.shift_left 1 id) (fun _ -> id); children = ch }
+
+let f () =
```

Comment:
Worth putting a comment here explaining this has to be in a function so that everything gets allocated on the heap rather than lifted.

Reply:
I also suggest `[@inline never]` on `f` to ensure this remains true under optimisations.